### PR TITLE
feat: add Classroom course cards

### LINKED ISSUE #86: Course card component for the cohort
### Feature description

Figma https://www.figma.com/file/KCfjq5xQHg0C1JmcrE3CGx/1332-Private-Experience?node-id=1420%3A105703

Screenshot 2022-11-03 at 8 28 29 AM

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes

Adds necessary queries to populate the Course cards on the Classroom page.

Resolves #86 

- Creates new Card component
- Adds new course service and store functions
- Adds carbon themes to be able to use theme tokens (eg `$text-primary`)

Does not include: 
- Store state reset - will follow up in separate PR #264 

## Implementation details

- Queries courses endpoint and checks the course admins field for current user
- Created basic card component. the course card currently does not link anywhere since the edit/create cohort flow is not completed. eventually it should drop users into the create/edit cohort flow

## How to read this PR

[Preview link](https://saiba-pr-260.tfb1j0wd7x2.us-south.codeengine.appdomain.cloud/account/classroom)

2. Login and go to classrooms page
3. You should not see any courses if you are not added as an admin to any courses
4. Log in to backend staging and add yourself to the [Private Course](https://api-dev.qiskit.org/admin/content/courses/0bb2b522-5705-4aec-9411-739517eabc1e) admin field 
5. Refresh classrom page. You should now see the admin course

## Screenshots

Screenshot 2022-10-31 at 10 07 49 AM

### CODE REVIEW

File: services/course.ts [hunk 1]
```diff
@@ -41,3 +41,36 @@ export const getCourse = (courseId: string, lang = 'en-US') => {
     (res: Query<CourseListDTO>) => res.data.courses[0]
   );
 };
+
+export const getCoursesByAdminID = (userID: string, language = 'en-US') => {
+  const query = /* GraphQL */ `
+    query GetCoursesByAdminID($id: String!, $language: String = "en-US") {
```
Comment (yoshiemuranaka) [inline services/course.ts:47]: Will need to change this depending on how we handle private/public/admin courses

File: services/course.ts [hunk 2]
```diff
@@ -41,3 +41,36 @@ export const getCourse = (courseId: string, lang = 'en-US') => {
...
+      courses(filter: { admins: { directus_users_id: { id: { _eq: $id } } } }) {
```
Comment (yoshiemuranaka) [inline services/course.ts:48]: Will need to change this depending on how we handle private/public/admin courses

With most recent backend changes, the admins field will be hidden

Comment (yoshiemuranaka) [reply services/course.ts:48]: We decided the admin field will be public, but the value will filter out user ids that are not the current user

**Comment by yoshiemuranaka** [comment]
This PR is ready for review

**Comment by Tansito** [comment]
Thanks @yoshiemuranaka tomorrow I will take a look 🙏

File: components/classroom/Courses.vue [hunk 3]
```diff
@@ -1,5 +1,74 @@
 <template>
-  <SectionContainer>
+  <bx-skeleton-text v-if="pending" type="heading" />
+  <SectionContainer v-if="courses.length">
     <h2>Courses</h2>
+    <div class="courses-container">
+      <Card
```
Comment (Tansito) [inline components/classroom/Courses.vue:6]: I see that `Card` for naming is quite generic, is this intentional? I see two options:
- `Card`: if we discussed that this design will be the design from now on for the `cards` in the platform.
- `<X>Card`: if we don't know if we are going to have different designs then I would be a bit more specific.

**Review by Tansito — APPROVED** [review]
I just left a little comment but I don't think it's blocking. Good job here @yoshiemuranaka thanks! 💯

File: components/classroom/Courses.vue (hunk 3 above)
Comment (Tansito) [reply components/classroom/Courses.vue:6]: For example, it's an option.

**Review by abdonrd — COMMENTED** [review]
Good job, @yoshiemuranaka! 👏🏻

File: store/course/store.ts [hunk 4]
```diff
@@ -99,8 +100,19 @@ export const useCourseStore = defineStore('courseStore', () => {
...
   }
 
+  function fetchAdminCourseData(userID: string) {
+    return useLazyAsyncData('fetch-course-by-admin-id', () =>
+      getCoursesByAdminID(userID)
+        .then((data) => (adminCourses.value = data.map(courseDTOToCourse)))
+        .catch((error) => {
+          throw error;
+        })
+    );
```
[the remark this answers is not in the document]
Comment (yoshiemuranaka) [reply store/course/store.ts:110]: I personally prefer promise chaining over try/catch blocks, but ok if async/await is preferred! The only thing I'm not sure about is the pending/error state returned to the component...I guess in the catch block throwing the error should work the same, I'll test this out!

File: components/classroom/Courses.vue (hunk 3 above)
Comment (yoshiemuranaka) [reply components/classroom/Courses.vue:6]: I initially thought I'd enhance this Card component to be used for syllabi cards and other use cases. The alternative I was thinking more recently is calling this a `MediaCard`. the syllabi cards have CTAs while cards with images/media probably will not have buttons in the card body, but the card itself would act as a clickable link. Then create another type of card that could be used for syllabi that would have primary/secondary buttons. 
what do you think?

File: store/types.ts [hunk 5]
```diff
@@ -6,3 +6,8 @@ export interface RequestState {
   pending: boolean;
   error: boolean | Error;
 }
+
+// https://docs.directus.io/reference/files.html#the-file-object
+export interface DirectusFileObject {
```
[the remark this answers is not in the document]
Comment (yoshiemuranaka) [reply store/types.ts]: Resolved by https://github.com/Qiskit/saiba/pull/260/commits/bd0ae2340e642ff6684f592b30f4c60f5556a02b

File: store/course/store.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (yoshiemuranaka) [reply store/course/store.ts:110]: Works well! and no need to wrap in try/catch the error bubble up and handled by the useLazyAsyncData hook

**Review by abdonrd — APPROVED** [review]
👏🏻

File: components/classroom/Courses.vue (hunk 3 above)
Comment (Tansito) [reply components/classroom/Courses.vue:6]: It has sense for me. My idea at least it's not to be so generic, if we think we can start closing components that's a good thing in fact!

Comment (yoshiemuranaka) [reply components/classroom/Courses.vue:6]: I'll make these changes in the next PR!

Comment (yoshiemuranaka) [reply components/classroom/Courses.vue:6]: Updated card component name here #290

**Pull request merged**